src/main/java/taub/gameoflife/RleReader.java

```diff
@@ -0,0 +1,152 @@
+                num.append(line.charAt(ix));
+                if (Character.isDigit(line.charAt(ix + 1)))
+                {
+                    num.append(line.charAt(ix + 1));
+                    ix = ix + 2;
+                } else {
+                    ix++;
+                }
+                if (line.charAt(ix) == 'o')
+                {
+                    for (int jx = 0; jx < Integer.parseInt(num.toString()); jx++)
+                    {
+                        grid.changeField(row, col);
+                        row++;
+                    }
+                } else if (line.charAt(ix) == 'b') {
+                    for (int jx = 0; jx < Integer.parseInt(num.toString()); jx++)
+                    {
+                            row++;
+                    }
+                }
+            }
+        }
+
+    }
+
+    // This method works when the pattern is row first
+    private void createPattern2(String line, int height)
+    {
+        for (int ix = 0; ix < line.length(); ix++)
+        {
+            if (!Character.isDigit(line.charAt(ix)))
+            {
+                if (line.charAt(ix) == 'o')
+                {
+                    grid.changeField(row, col);
+                    col++;
+                } else if (line.charAt(ix) == 'b') {
+                    col++;
+                } else if (line.charAt(ix) == '$') {
+                    row++;
+                    col = col - height;
+                }
+            } else {
+                StringBuilder num = new StringBuilder();
+                num.append(line.charAt(ix));
+                if (Character.isDigit(line.charAt(ix + 1)))
+                {
+                    num.append(line.charAt(ix + 1));
+                    ix = ix + 2;
+                } else {
+                    ix++;
+                }
+                if (line.charAt(ix) == 'o')
+                {
+                    for (int jx = 0; jx < Integer.parseInt(num.toString()); jx++)
+                    {
+                        grid.changeField(row, col);
+                        col++;
+                    }
+                } else if (line.charAt(ix) == 'b') {
+                    for (int jx = 0; jx < Integer.parseInt(num.toString()); jx++)
+                    {
+                            col++;
+                    }
+                }
+            }
+        }
+    }
+
+
+    private int getHeight(String line)
+    {
+        StringBuilder height = new StringBuilder();
+        for (int ix = 0; ix < line.length(); ix++)
+        {
+
+            if (Character.isDigit(line.charAt(ix)))
+            {
+                height.append(line.charAt(ix));
+                if (Character.isDigit(line.charAt(ix + 1)))
+                {
+                    height.append(line.charAt(ix + 1));
+                }
+                break;
+            }
+        }
+
+        int heightInt = Integer.parseInt(height.toString());
+        return heightInt;
+    }
```

Comment:
This isn't a great way to handle figuring out the height, you'll most likely have bugs with large patterns.
We can talk tomorrow about how best to do this.